services/compute/virtualmachine/virtualmachine.go

```diff
@@ -531,6 +553,42 @@ func (c *client) getVirtualMachineNetworkProfile(n *wssdcloudcompute.NetworkConf
 	return np
 }
 
+func (c *client) getVirtualMachineGuestAgentProfile(ga *wssdcloudcompute.GuestAgentConfiguration) *compute.GuestAgentProfile {
+	if ga == nil {
+		return nil
+	}
+
+	g := &compute.GuestAgentProfile{
+		Enabled: &ga.Enabled,
+	}
+
+	return g
+}
+
+func (c *client) getVirtualMachineGuestInstanceView(g *wssdcommon.VirtualMachineAgentInstanceView) *compute.GuestAgentInstanceView {
+	if g == nil {
+		return nil
+	}
+
+	gap := &compute.GuestAgentInstanceView{
+		AgentVersion: g.GetVmAgentVersion(),
+	}
+
+	for _, status := range g.Statuses {
```

Comment:
see this comment: https://github.com/microsoft/wssd-sdk-for-go/pull/217#discussion_r1231350290